In [ ]:
#while--->
#a.initialization
#b.condition
#c.increment and decrement

In [3]:
#question
i=1
n=4
while(i<=n):
    print(i)
    i=i+1

1
2
3
4


In [4]:
#question-
i=4
n=1
while(i>=n):
    print(i)
    i=i-1

4
3
2
1


In [ ]:
i=20#initialisation points
n=1
p=3 ### you can also write whether you want to skpi  1 2 or many steps
while(i>=n):###condition
    print(i)
    i=i-p

20
17
14
11
8
5
2


In [2]:
i=1
n=4
sum=0
while(i<=n):
    if i%2==0:
        print(i)
        sum=sum+i
        i=i+1
print(sum)
print("the sum is ",sum)

KeyboardInterrupt: 

In [ ]:
#question ---> write a program who print sum of square of give number
i=1
n=4
sum=0
while(i<=n):
    prod=i**2
    sum=sum+prod
    i=i+1
print(sum)


30


#step-1  
i=1,n=4,sum=0
while(1<=4)
prod=1*1=1
sum=0+1=1
now
i=2 ,sum=1

#step-2
i=2,n=4,sum=1
while(2<=4)
prod=2*2=4
sum=1+4=5
now 
1=3,sum=5
#step-3
i=3,sum=5,n=4
while(3<=4)
prod=3*3=9
sum=5+9=14
now 
i=4,sum=14

#step-4
i=4,n=4,sum=14
while(4<=4)
prod=4*4=16
sum=14+16=30
now
i=5,sum=30

#step-5
i=5,n=4,sum=30
while(5<=4)
break


In [ ]:
#question--> sum of cube of given number is armstrong(153=1**3+5**3+3**3=153)

i=153
orig=i
sum=0
while(i>0):
    sum=sum+(i%10)**3
    i=i//10
    print(sum)
if(orig==sum):
    print("armstrong number ")
else:
    print("not armstrong")

27
152
153


#step-1(i=153,sum=0)
while(i<0):
sum=0+(153%10)**3=0+3**3=27
153=153//10=15

#step-2(i=15,sum=27)
while(i>0):
sum=27+(15%10)**3=27+125=152
15//10=1

step-3(i=1,sum=152)
while(i>0):
sum=152+1
=153


In [19]:
i=1234
odd=0
even=0

while(i>0):
    d=i%10
    if(d%2==0):
        even=even+d
    else:
        odd=odd+d
    i=i//10
print("total even:",even,"total odd : ",odd)
        

total even: 6 total odd :  4


In [20]:
#question--->reverse of given number
i=1234
rev=0
while(i>0):
    rev=(rev*10)+(i%10)
    i=i//10
print(rev)

4321


#step-->(i=123,rev=0)
while(123>0):
rev=(0*10)+(123%10=3)=3
i=123//10=12

#step-->(i=12,rev=32)
while(12>0):
rev=(3*10)+(12%10=2)=32

#step-->(i=1,rev=321)
while(1>0)
rev=()


In [ ]:
#   QUESTION--->write a program who raturn given number is palindrom number

num=int(input("enter the number : "))
n=num
rev=0

while(num>0):
    rev=rev*10+(num%10)
    num=num//10
print(rev)
if n==rev:
    print("num is palindrom number")
else:
    print("not palindron number")

In [11]:
#QUESTION
num=5
i=1
prod=1
while(i<=10):
    prod=num*i
    print(f"{num}*{i}={prod}")
    i=i+1

5*1=5
5*2=10
5*3=15
5*4=20
5*5=25
5*6=30
5*7=35
5*8=40
5*9=45
5*10=50


In [21]:
num=84951
digit=0
largest_digit=0

while(num>0):
    digit=num%10
    if digit>largest_digit:
     largest_digit=digit
    else: 
       num=num//10
print(largest_digit)


9


In [27]:
num=1
count=0
sum=0
while(count<15):
    if num%2!=0:
        print(num)
        count=count+1
        sum=sum+num
    num=num+1
print(sum)   
    

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
225


In [28]:
#question-square of first 15 number even number
num=1
count=0
sum=0
while(count<15):
    if num%2==0:
        print(num)
        count=count+1
        num1=num**2
        sum=sum+num1
    num=num+1
print(sum) 



2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
4960


In [35]:
num=36
i=1

while(i<num):
    if num%i==0:
        print(i,end=" ")
    i=i+1

     
    

1 2 3 4 6 9 12 18 

In [2]:
text="python is fun"
text=text.split()
index=len(text)-1#because when we split its length will be means 3 we have to do 2 so len(text)-1
print(text[0])#just to check
n=" "
count=0
while count>0:
    print(text[index])
    result=text[index]
    index=index-1
    n =n + " " + result
    
print(n)


python
 


In [7]:
text = "python is fun"
rev = ""

index=len(text)-1

while index>=0:
    rev=rev+text[index]
    index=index-1
print(rev)


nuf si nohtyp


In [ ]:
str1 = "Python Program"
str1 = str1.lower() 

vowels = "aeiou"
vowel_count = 0
consonant_count = 0

index = 0
while index < len(str1):
    ch = str1[index]
    if ch.isalpha():      
        if ch in vowels:
            vowel_count += 1
        else:
            consonant_count += 1
    index += 1

print("Number of vowels:", vowel_count)
print("Number of consonants:", consonant_count)

Number of vowels: 3
Number of consonants: 10


In [ ]:
s="python program"
rev=""
index=len(s)-1
while index>=0:
    rev=rev+s[index]
    print(rev)
    index-=1
print(rev)

In [4]:
s="python program"
count=0
index=0

while index<len(s):
    if s[index]=='o':
        count=count+1
    index=index+1
print(count)



2


In [20]:
s="harshita"
n=s
rev=""
index=len(s)-1

while index>=0:
    rev=rev+s[index]
    index-=1
print(rev)
if rev==n:
    print("palindron string")
else:
    print("not palindrom string")

atihsrah
not palindrom string


In [24]:
s="p y t h o n"
index=0
new_s=""
while index<len(s):
    if s[index]!=" ":
        new_s=new_s+s[index]
    index=index+1
print(new_s)

python


In [29]:
s="python"
vowel="aeiou"
count=0
index=0
while index<len(s):
    if s[index] in vowel:
        count=count+1
    else:
        index=index+1
print(count)



KeyboardInterrupt: 